src/packages/apollo-client/src/index.ts

```diff
@@ -0,0 +1,33 @@
+const generateTypePolicyFields = (entityNames: string[]) => {
+	const policy = {
+		keyArgs: (
+			args: {
+				filter?: Record<string, unknown>;
+				pagination?: { orderBy: Record<string, unknown> };
+			} | null
+		) => {
+			// https://www.apollographql.com/docs/react/pagination/key-args/#keyargs-function-advanced
+			const filter = args?.filter ? JSON.stringify(args.filter) : '';
+			const orderBy = args?.pagination?.orderBy ? JSON.stringify(args.pagination.orderBy) : '';
+			return btoa(`${filter}:${orderBy}`);
+		},
+		merge(existing = [], incoming: { __ref: string }[]) {
+			const merged = [...existing, ...incoming];
+			const uniqueItems = new Set(merged.map((item) => item.__ref));
+			return [...uniqueItems].map((__ref) => merged.find((item) => item.__ref === __ref));
```

Comment:
So currently we're:

1. Creating a new array of everything in it
2. For looping that array to make a Set of refs
3. Making a new array of those refs, then for each of those, for looping the array in step 1 to look for it.

Couldn't we do this instead:
```typescript
const merged = new Map<Ref, Entity>();
for (const entity of [...incoming, ...existing]) {
    merged.set(entity.__ref, entity);
}
return merged.values();
```

Reply:
Updated this is much better!